src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -283,26 +274,28 @@ public static CallSite enumSwitch(MethodHandles.Lookup lookup,
             throw new IllegalArgumentException("Illegal invocation type " + invocationType);
         requireNonNull(labels);
 
-        labels = labels.clone();
-
         Class<?> enumClass = invocationType.parameterType(0);
-        labels = Stream.of(labels).map(l -> convertEnumConstants(lookup, enumClass, l)).toArray();
+        boolean constantsOnly = true;
+        int len = labels.length;
+
+        for (int i = 0; i < len; i++) {
+            Object convertedLabel =
+                    convertEnumConstants(lookup, enumClass, labels[i]);
+            labels[i] = convertedLabel;
+            constantsOnly &=
+                    EnumDesc.class.isAssignableFrom(convertedLabel.getClass());
```

Comment:
```suggestion
            if (constantsOnly)
                constantsOnly = EnumDesc.class.isAssignableFrom(convertedLabel.getClass());
```

`&=` does not short-circuit the `isAssignableFrom` evaluation.

Reply:
It’d have to be `&&=`, but I don’t think **Java** supports that.